generator/src/main/java/com/scottlogic/deg/generator/generation/fieldvaluesources/CombiningFieldValueSource.java

```diff
@@ -52,11 +54,15 @@ public CombiningFieldValueSource(List<FieldValueSource> underlyingSources) {
 
     @Override
     public Stream<Object> generateRandomValues(RandomNumberGenerator randomNumberGenerator) {
-        return stream(new InternalRandomIterator(
-            underlyingSources.stream()
-                .map(source -> source.generateRandomValues(randomNumberGenerator).iterator())
-                .collect(Collectors.toList()),
-            randomNumberGenerator));
+        return flatMap(
+            Stream.generate(() -> getRandomSource(randomNumberGenerator)
+                .generateRandomValues(randomNumberGenerator)
+                .limit(1)),
```

Comment:
Can we talk about this? I am a bit confused as to why this is here.

Reply:
limit(1) instead of findFirst required so null can be returned